specs/Picker-API_spec.md

```diff
@@ -0,0 +1,116 @@
+   files. It does not however, allow developers to specify a mode to pick both files and folders.
+   There has been feedback indicating that this behaviour is limiting, requiring developers
+   to build their own picker experience. 
+
+**2. Unable to pick multiple folders.**
+
+   The Picker dialog (FolderPicker) does not allow users to pick multiple folders. The developer
+   experience currently is to select a single folder and repeat the dialog for more folders. This 
+   is a less than ideal experience for application developers.
+
+The Reunion picker dialog is going to address the above issues. This Reunion API will be based upon the below
+APIs in the Windows SDK. The Reunion Picker API will remove the deprecated APIs and add additional methods
+for the functionality: File and folder picker, Multiple folder picker.
+
+- Windows.Storage.Pickers.FileOpenPicker (https://docs.microsoft.com/en-us/uwp/api/Windows.Storage.Pickers.FileOpenPicker)
+- Windows.Storage.Pickers.FileSavePicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.filesavepicker)
+- Windows.Storage.Pickers.FolderPicker (https://docs.microsoft.com/en-us/uwp/api/windows.storage.pickers.folderpicker) 
+
+# Description
+
+The goal of this API is to address the gaps in the existing File/Folder picker APIs. The API surface is similar to the existing
+Picker APIs in Windows SDK with additional methods to support File + Folder picker and multiple folder picker. The API removes the
+deprecated methods from Windows SDK.
+
+# Examples
+
+## Show file and folder picker
+    ```c#
+    var picker = StorageItemPicker::CreateFilePicker();
+    picker.Filters.add("*.jpg");
+    picker.Filters.add("*.png");
+    picker.StartLocation = startLocation;
+    IReadOnlyList<StorageItem> storageItems = await picker.PickStorageItemAsync();
+
+    if(0 < storageItems.Count)
+    {
+    }
+    ```
+
+## Show single folder picker
+    ```c#
+    var picker = StorageItemPicker::CreateFolderPicker();
+    Windows.Storage.StorageFolder folder = await picker.PickSingleFolderAsync();
+    ```
+
+
+## Show multi folder picker
+    ```c#
+    var picker = StorageItemPicker::CreateFolderPicker();
+    IReadOnlyList<Windows.Storage.StorageFolder> folders = picker.PickMultipleFoldersAsync();
+    ```
+
+# Remarks
+
+# API Notes
+
+
+# API Details
+
+    ```c#
+    namespace Microsoft.Reunion.Storage.Pickers
+    {
+        typedef enum PickerViewMode
+        {
+            List = 0,
+            Details,
+            Tiles,
+            Content
+        } PickerViewMode;
+
+        runtimeclass StorageItemPicker
+        {
+            /// Static constructor
+            static FileOpenPicker CreateFileOpenPicker();
+            static FileSaveFileAsync CreateFileSavePicker();
+            static FolderPicker CreateFolderPicker();
+
+            String StartLocation{ set; }
+            String OkButtonText { set; }
```

Comment:
The description said that this API was just additions and removal of deprecations, but this is a rename (current is CommitButtonText).

Reply:
The API is not just additions and removal of deprecations. The API is going to be a re-write of the picker APIs since the one in the current form is not very useful. The use of the API requires some amount of rewrite of the existing code.